RetailCoder.VBE/UI/SelectionChangeService.cs

```diff
@@ -123,7 +123,7 @@ private void DispatchSelectedDesignerDeclaration(IVBComponent component)
                 return;
             }
             var form =
-                _parser.State.DeclarationFinder.UserDeclarations(DeclarationType.UserForm).SingleOrDefault(decl =>
```

Comment:
`UserForm` wasn't working?

Reply:
Nope, parser just flags them as standard ClassModules, despite the bitwise.

Reply:
nah, all good.

Reply:
Something isn't working as intended: [UserDeclarations(DeclarationType) *means* to check for a flag](https://github.com/rubberduck-vba/Rubberduck/blob/64860377a5c5057a27e62bce5a56dc249838e7bf/Rubberduck.Parsing/Symbols/DeclarationFinder.cs#L377):

        public IEnumerable<Declaration> UserDeclarations(DeclarationType type)
        {
            List<Declaration> result;
            return _userDeclarationsByType.TryGetValue(type, out result)
                ? result
                : _userDeclarationsByType
                    .Where(item => item.Key.HasFlag(type))
                    .SelectMany(item => item.Value);
        }

Reply:
Kind of. As far as I can see, it uses all types with the flag only if there is no declaration with the exact type. 
Moreover, the `UserForm` has the `ClassModule` flag and not the other way around. So, `UserDeclarations(DeclarationType.UserForm)` will never find user forms with type `ClassModule`.

Reply:
see https://chat.stackexchange.com/transcript/message/40183276#40183276